datastream/dataset.py

```diff
@@ -87,6 +87,17 @@ def from_dataframe(dataframe: pd.DataFrame) -> Dataset[pd.Series]:
             get_item=lambda df, index: df.iloc[index],
         )
 
+    def new_columns(self, **kwargs):
+        '''
+        Appends new column(s) to ``self.dataframe`` using
+        ``pd.DataFrame.assign``.
+        '''
+        return Dataset(
+            dataframe=self.dataframe.assign(**kwargs),
+            length=len(self),
```

Comment:
I would prefer to use the same creation method + map

`Dataset.from_dataframe(self.dataframe).map(self.get_item)`

Reply:
Not sure this would work since the `self.get_item` created by `from_dataframe` returns `self.dataframe[index]` which the original `self.get_item` probably doesn't accept as input.

Reply:
This does not work